packages/react-mutation-mapper/src/component/track/TrackRect.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react';
+import { action, computed, makeObservable, observable } from 'mobx';
+import { observer } from 'mobx-react';
+
+type TrackRectProps = {
+    x: number;
+    y: number;
+    width?: number;
+    height?: number;
+    isHovered: boolean;
+    hitZoneClassName?: string;
+    hitZoneXOffset?: number;
+    spec: TrackRectSpec;
+};
+
+export type TrackRectSpec = {
+    startCodon: number;
+    endCodon?: number;
+    label?: string;
+    labelColor?: string;
+    color?: string;
+    tooltip?: JSX.Element;
+};
+
+@observer
+export default class TrackRect extends React.Component<TrackRectProps, {}> {
+    @observable private textElt: SVGTextElement | null = null;
+    private handlers: any;
+    constructor(props: any) {
+        super(props);
+        makeObservable(this);
+        this.state = {
+            displayText: props.label || '',
+        };
+        this.handlers = {
+            textRef: action((text: SVGTextElement | null) => {
+                this.textElt = text;
+            }),
+        };
+    }
+
+    public static defaultProps = {
+        width: 50,
+        height: 10,
+    };
+
+    private get centerX() {
+        return this.props.x + this.props.width! / 2;
+    }
+
+    private get centerY() {
+        return this.props.y + this.props.height! / 2;
+    }
+
+    @computed private get displayText() {
```

Comment:
This is almost identical to `Domain.displayText` method. Might be better to add a utility function and use it in both places.